forms-flow-data-layer/src/graphql/service/formio_form_service.py

```diff
@@ -106,3 +110,84 @@ async def get_submissions(submission_id: str):
         return (
             submission.data if submission else None
         )  # Return data if found, otherwise None
+
+    @staticmethod
+    async def _build_match_stage(submission_ids: List[str], search: Optional[dict]) -> dict:
+        """Build the MongoDB match stage."""
+        match_stage = {"_id": {"$in": [ObjectId(id) for id in submission_ids]}}
+        if search:
+            for field, value in search.items():
+                match_stage[f"data.{field}"] = value
```

Comment:
we need to implement or condition and also value operator is contains after merge this pr we can do that